src/createAsyncStoragePersister/asyncThrottle.ts

```diff
@@ -0,0 +1,51 @@
+import { getLogger } from '../core/logger'
+
+export function asyncThrottle<Args extends readonly unknown[]>(
+  func: (...args: Args) => Promise<void>,
+  interval = 1000
+) {
+  if (typeof func !== 'function') throw new Error('argument is not function.')
+
+  let running = false
+  let lastTime = 0
+  let timeout: ReturnType<typeof setTimeout>
+  let currentArgs: Args | null = null
+
+  const execFunc = async () => {
+    if (currentArgs) {
+      const args = currentArgs
+      currentArgs = null
+      try {
+        running = true
+        await func(...args)
+      } catch (error) {
+        getLogger().error(error)
+      } finally {
```

Comment:
one small thing: in v4, we put all loggings behind a production env check, like here:

https://github.com/tannerlinsley/react-query/blob/fe98d533cc62fab8ab88166a31707d6cfc02d058/src/core/query.ts#L449-L451

can we do that here as well please

Reply:
oh but actually, I don't think we should be logging here on this level. We have [another PR](https://github.com/tannerlinsley/react-query/pull/3246) where we removed the global logger and made it an option on the queryClient instead, so this would be problematic.

If we want to catch errors and log them, how about doing it in the `persistQueryClient` itself, when the writing happens? I think this would be somewhere around here:

https://github.com/tannerlinsley/react-query/blob/1abb3714637c3f9044a4294cc737cc41966a2f49/src/persistQueryClient/index.ts#L114

Not sure if it makes sense to just swallow and log the error though. Shouldn't we at least re-throw it to the consumers?